server.cpp

```diff
@@ -1,25 +1,56 @@
 #include "FileGopherServer.h"
+#include "boost/program_options.hpp"
 
 using namespace std;
+namespace po = boost::program_options;
 
 // Driver function.
 int main(int argc, char *argv[])
 {
-    // Print Usage.
-    if (argc != 2)
+    // variables needed to be captured from command line options
+
+    string directory = "."; // defaults to cwd
+    int port = 70;          // default port value of 70
+    bool show_hidden_files = false;
+
+    // Declare the supported options.
+    po::options_description desc("Allowed options");
+    desc.add_options()
+    ("directory,d", po::value<string>(), "the directory to serve")
+    ("port,p", po::value<int>(), "the port to serve on (default : 70)")
+    ("hidden,h", po::value<bool>(), "show hidden files if specified");
+
+    po::variables_map vm;
+    po::store(po::parse_command_line(argc, argv, desc), vm);
+    po::notify(vm);
+
+    if (vm.count("help"))
     {
-        printf("USAGE: \n\tserver <directory-to-serve>\n\n");
-        printf("Where <directory-to-serve> root directory of server.\n");
-        return 0;
+        cout << desc << "\n";
+        return 1;
     }
+
+    if (vm.count("compression"))
```

Comment:
Should this be `compression` or `directory`?

Reply:
This is really bad, sorry lol :)